pilot/pkg/xds/delta.go

```diff
@@ -160,7 +160,7 @@ func (s *DiscoveryServer) pushConnectionDelta(con *Connection, pushEv *Event) er
 		s.computeProxyState(con.proxy, pushRequest)
 	}
 
-	if !s.ProxyNeedsPush(con.proxy, pushRequest) {
+	if !s.ProxyNeedsPush(con.proxy, pushRequest) || (features.EnableConditionalXDSPush && pushRequest.SkipXDSPush) {
```

Comment:
Might be good to have a special log line so we know if the skip was due to gateway api status changing only